### PR TITLE
Revert package type to CommonJS

### LINKED ISSUE #721: Error importing trino-client because of type mismatch
The trino-client configuration is inconsistent in 0.2.4, and this mismatch is causing an error. 

```
const trino_client_1 = require("trino-client");
                       ^

Error [ERR_REQUIRE_ESM]: require() of ES Module .../Documents/lightdash/node_modules/trino-client/dist/index.js from .../Documents/lightdash/packages/warehouses/dist/warehouseClients/TrinoWarehouseClient.js not supported.
index.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead either rename index.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in .../Documents/lightdash/node_modules/trino-client/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
```

### Here's why:

`type: "module" `in package.json:
This tells Node.js to treat all .js files in the package as ES Modules by default. This affects how Node.js resolves and executes the files.

`"module": "commonjs"` in tsconfig.json:
This tells TypeScript to compile the code as CommonJS modules. The emitted files will use CommonJS syntax (e.g., require instead of import).

### Why is This Problematic?
When type: "module" is set:

Node.js expects the files to use ES Module syntax (import/export).
But the compiled CommonJS code (require/module.exports) doesn't match this expectation.
This creates conflicts because Node.js cannot correctly interpret CommonJS output in a package marked as an ES Module.

### Solution

I don't know why `type: "module"` was recently added to package.json but the simplest solution is to rollback this change.
Note that 0.2.3 works fine but we would like to use the latest version to patch a [security vulnerability in axious](https://security.snyk.io/vuln/SNYK-JS-AXIOS-7361793) which is a dependency from trino-client.

### PR DESCRIPTION
**Pull request by regadas**
Fixes #721

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "version": "0.2.4",
   "name": "trino-client",
   "description": "Trino client library",
-  "type": "module",
```
Comment (nineinchnick) [inline package.json:5]: You need to also update `eslint.config.js` to something like:
```
module.exports = [{
    ignores: ["dist/*"],
}]
```
or just copy the default, that looks like: https://github.com/trinodb/github-actions/blob/main/action-surefire-report/eslint.config.cjs

Comment (regadas) [reply package.json:5]: can you take a look at #720 first?

Comment (regadas) [reply package.json:5]: This will work as expected afterwards

Comment (nineinchnick) [reply package.json:5]: Please rebase and LGTM once the CI is green.

**Review by nineinchnick — APPROVED** [review]

**Pull request merged**